manifests/windows/install.pp

```diff
@@ -56,4 +56,11 @@
     command => "${::system32}\\cmd.exe /c start /b ${_cmd_location} /c \"${_installbat}\" ${::puppet_agent_pid}",
     path    => $::path,
   }
+
+  # PE-15037 Cache dir loses inheritable SYSTEM perms
+  exec { 'fix inheritable SYSTEM perms':
+    command => "${::system32}\\icacls.exe ${::puppet_client_datadir} /grant \"SYSTEM:(OI)(CI)(F)\"",
```

Comment:
On older versions of windows this path could contains spaces, so I think you want to wrap the path in double quotes to be safe like:

```
"${::system32}\\icacls.exe \"${::puppet_client_datadir}\" /grant \"SYSTEM:(OI)(CI)(F)\""
```